src/main/java/se/liu/ida/hefquin/engine/queryproc/impl/optimizer/simple/RandomizedJoinPlanOptimizerImpl.java

```diff
@@ -0,0 +1,50 @@
+package se.liu.ida.hefquin.engine.queryproc.impl.optimizer.simple;
+
+import java.util.List;
+import java.util.Random;
+
+import se.liu.ida.hefquin.engine.queryplan.PhysicalPlan;
+import se.liu.ida.hefquin.engine.queryplan.utils.PhysicalPlanFactory;
+import se.liu.ida.hefquin.engine.queryproc.QueryOptimizationException;
+
+public class RandomizedJoinPlanOptimizerImpl implements JoinPlanOptimizer {
+	
+	/**
+	 * The class contains a random seed object.
+	 * This cannot be marked final (compilation error if so).
+	 */
+	protected Random random = new Random();
+	
+	/**
+	 * Have a list of subplans.
+	 */
+	protected final List<PhysicalPlan> subplans;

```

Comment:
You should not have this member variable. The class should be stateless. In fact, you are not even using this member variable in the `determineJoinPlan` function (because there you are using the function argument with the same name). Hence, please remove this member variable, which means that you also need to remove it from the constructor.

After removing it from the constructor, and removing the part about `this.random` from the constructor as well (see my comment there), the whole constructor becomes empty and, thus, can and should be removed altogether.